File: Problems/No.1248_Count_Number_of_Nice_Subarrays/count_number_of_nice_subarrays.py
class Solution(object):
    def numberOfSubarrays(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """

        # parameters
        subarray_start = -1
        odd_IDs = []
        nice_count = 0

        # start computing
        for i in range(len(nums)):
            if (nums[i]%2 == 1):
                odd_IDs.append(i)
            if (len(odd_IDs) == k):
                nice_count += odd_IDs[0] - subarray_start
            elif (len(odd_IDs) > k):
                subarray_start = odd_IDs.pop(0)
                nice_count += odd_IDs[0] - subarray_start
        
        return nice_count

File: Problems/No.1248_Count_Number_of_Nice_Subarrays/count_number_of_nice_subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        
        // parameter
        int subarray_start = 0;
        int temp_count = 0;
        int nice_count = 0;

        // start computing
        for (int i=0; i<nums.size(); i++) {

            if (nums[i]%2 == 1) {

                k--;
                if (k <= 0) {

                    temp_count = 1;
                    while (nums[subarray_start++]%2 == 0) temp_count++;
                    nice_count += temp_count;    

                }

            }
            else nice_count += temp_count;

        }

        return nice_count;

    }
};
